### ISSUE TITLE
PHP $SERVER_PORT empty when port specified on command-line

### ISSUE THREAD

**Issue by 4nd** [mholt/caddy#888]
(Are you asking for help with using Caddy? Please use our forum instead: https://forum.caddyserver.com. If you are filing a bug report, please answer the following questions. If your issue is not a bug report, you do not need to use this template. Either way, please consider donating if we've helped you. Thanks!)

#### 1. What version of Caddy are you running (`caddy -version`)?
Caddy 0.8.3

#### 2. What are you trying to do?
SERVER_PORT in PHP is empty, need it to contain the port the server is running on

#### 3. What is your entire Caddyfile?
```text
0.0.0.0 {

    root public
    errors error.log

    fastcgi / /opt/local/var/run/php56/php5-fpm.sock php {
        env OTHER_PORT {$PORT}
    }
}
```

#### 4. How did you run Caddy (give the full command and describe the execution environment)?
command: caddy --port=50000 
os x

#### 5. What did you expect to see?
In the phpinfo() output $_SERVER['SERVER_PORT'] and $_SERVER['OTHER_PORT'] have no value
if I specify the port in the Caddyfile e.g. 0.0.0.0:50000 the values are set correctly '50000'

#### 6. What did you see instead (give full error messages and/or log)?
I expect to see the $_SERVER['SERVER_PORT'] or anything else using {$PORT} set based on the passed in port number at the command-line

#### 7. How can someone who is starting from scratch reproduce this behavior as minimally as possible?
execute a config with php fastcgi as above and run php_info.  Run Caddy at command-line and specify a port, its not passed through to php SERVER_PORT

**Comment by abiosoft** [comment]
Seems you are reporting two issues in one.

1. `{$PORT}` being empty.
If you want `$PORT` environment variable, then you should have set it for the environment before starting Caddy.

2. `$_SERVER['SERVER_PORT']` being empty when you pass port via cli flag.
Considering the value is not empty when you specify port in the Caddyfile instead, I will assume this was missed out somewhere and should not be difficult to fix.

TLDR. 1 is not an issue. 2 may be a bug.

**Comment by 4nd** [comment]
Is {$PORT} (or {%PORT%}) not a standard variable that can be used in the Caddyfile?  the line 
``text
env OTHER_PORT {$PORT}
```
does populate $_SERVER['OTHER_PORT'] if I specify a port at the vhost level -  0.0.0.0:50000.

So I suspect the {$PORT} variable just isn't being set as you suggested in your point 2. That is the bug I was reporting.  

Apologies for the confusion.

**Comment by mholt** [comment]
> I expect to see the $_SERVER['SERVER_PORT'] or anything else using {$PORT} set based on the passed in port number at the command-line

I think this is the confusion: the port passed in at the command line is not an environment variable, you want to run `export PORT=50000` before you run caddy; that will make it an environment variable. (But that alone will not change the value of the `-port` flag; you'd need `-port=$PORT` to do that.)

**Comment by 4nd** [comment]
Hi Matt,

When I say environment variable I mean it as a $_SERVER variable in PHP, not a shell environment variable.  So lets forget about environment variables for now, they're a symptom of the problem.

When I specify the port in the Caddyfile using something like 127.0.0.1:50000 as the host identifier, the PHP value of $_SERVER['SERVER_PORT'] is correctly set as 50000.

If I do not specify the port in the Caddyfile e.g. 127.0.0.1 but use the command-line -port flag instead e.g. -port=50000 it does not pass through to PHP and the value of $_SERVER['SERVER_PORT'] is empty. It should never be empty.  It should always be populated with the port on which the web server is running.

Does that make better sense, I think I confused everything with the environment variable terminology.

**Comment by mholt** [comment]
Okay, yes, that makes much more sense :smile: - so it actually has nothing to do with `{$PORT}`. Got it!

Will look into it from that angle now.
